Q1. What are the three measures of central tendency?

Measures of central tendency help you find the middle, or the average, of a dataset. The 3 most common measures of central tendency are the mode, median, and mean.
Mode: the most frequent value.
Median: the middle number in an ordered dataset.
Mean: the sum of all values divided by the total number of values.
In addition to central tendency, the variability and distribution of your dataset is important to understand when performing descriptive statistics.

Q2. What is the difference between the mean, median, and mode? How are they used to measure the
central tendency of a dataset?

Mean:-	
1.The average taken for a set of numbers is called a mean.
2.Add all of the numbers together and divide the sum by the total number of values.
3.The result is the mean or average score.
4.Example: To find the average of the four numbers 2, 4, 6, and 8, we need to add the number first.2 + 4 + 6+ 8 = 20
Divide the sum by the total number of numbers, i. e 4.
20/4 = 5 is the average or mean

Median:-
1.The middle value in the data set is called the Median.
2.Place all the given numbers in an ascending order
3.The next step is to find the middle number on the list. It is called the median.
4.Example: 4, 2, 8, 10, 19.Arrange the numbers in ascending order. i .e., 2, 4, 8, 10, 19.
As the total numbers are 5, so the middle number 8 is the median here.

Mode:-
1.The number that occurs the most in a given list of numbers is called a mode.
2.It shows the frequency of occurrence.
3.We can have more than one mode or no mode at all.
4.Example: 3, 3, 5, 6, 7, 7, 8, 1, 1, 1, 4, 5, 6.Find the frequency of each number.
For number 3, it’s 2. For 5, it’s 2. For 6, it’s 2. For 7, it’s 2. For 8, it’s one. For 1, it’s 3. For 4, it’s 1.
The number with the highest frequency is the mode. Hence, the mode of the given sequence of numbers is 1.

Q3. Measure the three measures of central tendency for the given height data:[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [3]:
import numpy as np

In [2]:
heights=[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [3]:
np.mean(heights)

177.01875

In [4]:
np.median(heights)

177.0

In [6]:
from scipy import stats
stats.mode(heights)

/tmp/ipykernel_378/2713150148.py:2: FutureWarning: Unlike other reduction functions (e.g. `skew`, `kurtosis`), the default behavior of `mode` typically preserves the axis it acts along. In SciPy 1.11.0, this behavior will change: the default value of `keepdims` will become False, the `axis` over which the statistic is taken will be eliminated, and the value None will no longer be accepted. Set `keepdims` to True or False to avoid this warning.
  stats.mode(heights)


ModeResult(mode=array([177.]), count=array([3]))

Q4. Find the standard deviation for the given data:
[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [7]:
data=[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [8]:
np.std(data)

1.7885814036548633

Q5. How are measures of dispersion such as range, variance, and standard deviation used to describe
the spread of a dataset? Provide an example.

Measures of Dispersion
The measures of dispersion contain almost the same unit as the quantity being measured. There are many Measures of Dispersion found that help us to get more insights into the data: 
1.Range 
2.Variance 
3.Standard Deviation 
4.Skewness 
5.IQR  
Range:-
Range is the measure of the difference between the largest and smallest value of the data variability. The range is the simplest form of Measures of Dispersion
Example: 1,2,3,4,5,6,7 
Range = Highest value – Lowest value 
  = ( 7 – 1 ) = 6 
Variance (σ2):-
In simple terms, the variance can be calculated by obtaining the sum of the squared distance of each term in the distribution from the Mean, and then dividing this by the total number of the terms in the distribution.It basically shows how far a number, for example, a student’s mark in an exam, is from the Mean of the entire class.
Standard Deviation:-
Standard Deviation can be represented as the square root of Variance. To find the standard deviation of any data, you need to find the variance first. Standard Deviation is considered the best measure of dispersion.Standard Deviation = √σ 

Q6. What is a Venn diagram?

Venn diagrams are visual representations of mathematical sets—or collections of numbers or things—that are studied using a branch of logic called set theory. Set theory is one of the foundational systems for mathematics, and it helped to develop our modern understanding of infinity and real numbers.Researchers and mathematicians have developed a language and system of notation around set theory. If you want to get in on their secrets, you'll want to become familiar with these Venn diagram symbols.

Q7. For the two given sets A = (2,3,4,5,6,7) & B = (0,2,6,8,10). Find:

In [14]:
A={2,3,4,5,6,7}
B={0,2,6,8,10}

(i) A ⋂ B

In [15]:
A.intersection(B)

{2, 6}

(ii) A ⋃ B

In [16]:
A.union(B)

{0, 2, 3, 4, 5, 6, 7, 8, 10}

Q8. What do you understand about skewness in data?

Skewed data is data that creates an asymmetrical, skewed curve on a graph. In statistics, the graph of a data set with normal distribution is symmetrical and shaped like a bell. However, skewed data has a "tail" on either side of the graph. The two most common types of skew are:
Negative skew: A data set with a negative skew has a tail on the negative side of the graph, meaning the graph is skewed to the left.
Positive skew: A data set with a positive skew has a tail on the positive side of the graph, meaning the graph is skewed to the right.
Positive skew example:-
One example of positively skewed data could be a typical income data set. If you draw a curve of a sample population's income on a graph, the curve is likely to be skewed to the right, or positively skewed. This would occur if most people have average incomes, and a smaller number of people have high incomes. The people with high incomes would be the outliers in the data set that skew the curve toward the right side of the graph.

Q9. If a data is right skewed then what will be the position of median with respect to mean?

When the distribution is skewed to the right, the mean is often greater than the median. In symmetric distributions, we expect the mean and median to be approximately equal in value. This is an important connection between the shape of the distribution and the relationship of the mean and median.

Q10. Explain the difference between covariance and correlation. How are these measures used in
statistical analysis?

Covariance:-
1.Covariance is an indicator of the extent to which 2 random variables are dependent on each other. A higher number denotes higher dependency.
2.The value of covariance lies in the range of -∞ and +∞.
3.Change in scale Affects covariance
4.Unit free measure No
Correlation:-
1.Correlation is a statistical measure that indicates how strongly two variables are related.
2.Correlation is limited to values between the range -1 and +1
3.Change in scale Does not affect the correlation
4.Unit free measure Yes
Covariance is an indicator of how two random variables change concerning each other. Correlation, on the other hand, measures the strength of this relationship.

Q11. What is the formula for calculating the sample mean? Provide an example calculation for a
dataset.

The sample mean formula is: x̄= (xi)/n It is easier than you believe if that sounds complex. Remember the formula in basic math to find an average.

In [20]:
import seaborn as sns

In [18]:
df=sns.load_dataset("tips")

In [19]:
df.head()

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4

In [21]:
np.mean(df["tip"])

2.99827868852459

Q12. For a normal distribution data what is the relationship between its measure of central tendency?

Mean:-The arithmetic mean of a dataset (which is different from the geometric mean) is the sum of all values divided by the total number of values. It’s the most commonly used measure of central tendency because all values are used in the calculation.
median  of even dataset:-
For an even-numbered dataset, find the two values in the middle of the dataset: the values at the \dfrac{n}{2} and (\dfrac{n}{2})+1 positions. Then, find their mean.
median for odd dataset;-
For an odd-numbered dataset, find the value that lies at the \dfrac{(n+1)}{2} position, where n is the number of values in the dataset.
Mode:-
The mode is the most frequently occurring value in the dataset. It’s possible to have no mode, one mode, or more than one mode.
To find the mode, sort your dataset numerically or categorically and select the response that occurs most frequently.

Q13. How is covariance different from correlation?

Put simply, both covariance and correlation measure the relationship and the dependency between two variables. Covariance indicates the direction of the linear relationship between variables while correlation measures both the strength and direction of the linear relationship between two variables. Correlation is a function of the covariance. 
What sets these two concepts apart is the fact that correlation values are standardized whereas covariance values are not. 
You can obtain the correlation coefficient of two variables by dividing the covariance of these variables by the product of the standard deviations of the same values.
Don’t forget that standard deviation measures the absolute variability of a data set’s distribution. When you divide the covariance values by the standard deviation, it essentially scales the value down to a limited range of -1 to +1. This is precisely the range of the correlation values.

Q14. How do outliers affect measures of central tendency and dispersion? Provide an example.?

When a outlier is present it can effect the shape of the graph, if we have outliers to the right of the graph. These outliers are causing the mean to increase, but if we have outliers to the left of the graph these outliers are dragging down the mean. This causes a conflict because the mean no longer provides a good representation of the data, alternatively we would much rather use the median. The median on the other hand is less likely to be affect by outliers.

In [1]:
lis=[1,2,3,4]

In [4]:
np.mean(lis)

2.5

In [5]:
#outlier is 200
lis.append(200)

In [6]:
np.mean(lis)

42.0

Hence,as you can se above because of 200 the mean increased from 2.5 to 42.0 directly.